src/cpp/server/server_cc.cc

```diff
@@ -410,11 +415,27 @@ class Server::CallbackRequest final : public internal::CompletionQueueTag {
       GPR_ASSERT(ignored == req_);
 
       if (!ok) {
-        // The call has been shutdown
-        req_->Clear();
+        // The call has been shutdown. Let it stay in the list
+        // since the server shutdown will delete and Clear it.
+        req_->Done();
```

Comment:
Why do not we need to clear the request any more?

Reply:
The destructor clears it. I'm also going to change this code based on Sheena's comment as well.